src/extension.ts

```diff
@@ -4,16 +4,28 @@ import { identify } from "./exercise/identification";
 import { Submission, SubmissionResponse } from "./submission";
 import { AssertionError } from 'assert';
 
+// Get the API token from the settings.
+let config = vscode.workspace.getConfiguration('dodona');
 const get = bent('json');
 const getHTML = bent();
-const post = bent('https://dodona.ugent.be', 'POST', 'json');
 const sleep = (amt: number) => new Promise(r => setTimeout(r, amt));
 
 export function activate(context: vscode.ExtensionContext) {
     const disp = vscode.commands.registerCommand('extension.submit', async () => {
         // Get the current editor.
         const editor = vscode.window.activeTextEditor;
         if (editor) {
+            // Get the configuration from the settings.
+            // Re-declare in case the user changed it.
+            config = vscode.workspace.getConfiguration('dodona');
+
+            // Declare this here so that the platform can be changed without restarting.
+            // Cast to lowercase so the enum values can be CamelCased to look better.
+            // Error can be ignored as this configuration has a default value & is always a string.
+            //@ts-ignore
+            const domain = config.get("platform").toLowerCase();
+            const post = bent(`https://${domain}.ugent.be`, 'POST', 'json');
```

Comment:
By specifying the host in the settings directly (see previous remark), the domain does not need to be lowercased and can be used directly.
**Make sure to test if it still works, it's possible that `@ts-ignore` is still required anyway**

```suggestion
            const post = bent(config.get('api.host'), 'POST', 'json');
```